src/datasets/arrow_writer.py

```diff
@@ -433,7 +433,13 @@ def write_examples_on_file(self):
         batch_examples = {}
         for col in cols:
             # Since current_examples contains (example, key) tuples
-            batch_examples[col] = [row[0][col] for row in self.current_examples]
+            if isinstance(self.current_examples[0][0][col], (pa.Array, pa.ChunkedArray)):
```

Comment:
```suggestion
            if isinstance(self.current_examples[0][0][col], (pa.Array, pa.ChunkedArray)):
                # The examples could be Arrow arrays of 1 element.
                # This can happen in `.map()` when we want to re-write the same Arrow data
```